.gitignore

```diff
@@ -1,3 +1,14 @@
 # Ignore (optional) configuration files with user-specific paths:
 initial_cache.cmake
 setup.cfg
+
+# Mac specific files
+*.DS_Store
+
+# IntelliJ IDEA files
```

Comment:
Regarding the IntelliJ files, is it not possible to do out-of-source builds what that IDE? We prefer to recommend that users to out-of-source builds, instead of putting build directories into the source directory and then adding those build directories to `.gitignore`.
Of course if it's unavoidable IDE project files, we could add it here.

Reply:
Clion IDE creates a hidden folder `.idea/` in the root of the project. Authors of this IDE recommend to add it in `.gitignore` (https://intellij-support.jetbrains.com/hc/en-us/articles/206544839).

What is about `cmake-build-*` - it is just build folder and yes, it seems reasonable to remove it from here.

Reply:
Done